apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nav-sidebar/class-wpcom-block-editor-nav-sidebar.php

```diff
@@ -95,7 +95,8 @@ public function enqueue_script_and_style() {
 	 */
 	public function add_wpcom_dashboard_link( $settings ) {
 		$site_slug                               = preg_replace( '|^https?:\/\/|', '', home_url() );
-		$settings['__experimentalDashboardLink'] = 'https://wordpress.com/home/' . $site_slug;
+		$site_hostname                           = ! empty( $_GET['origin'] ) ? $_GET['origin'] : 'https://wordpress.com'; // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		$settings['__experimentalDashboardLink'] = $site_hostname . '/home/' . $site_slug;
```

Comment:
Quick thoughts - The code seems reasonable, but the location is a bit suspect (noted this already existed here before your change). This sidebar in ETK is only used in the post editor, while we are registering this plugin and setting up this setting might work for where this `__experimentalDashboardLink` is consumed in the site editor, its a bit odd that we are defining it here in this other sidebar plugin.